src/util/log.h

```diff
@@ -14,6 +14,13 @@
 #define ASSERT(pass, msg, ...) _smek_assert(__FILE__, __LINE__, __func__, pass, STR(pass), msg, ##__VA_ARGS__)
 #define CHECK(pass, msg, ...)  _smek_check(__FILE__, __LINE__, __func__, pass, STR(pass), msg, ##__VA_ARGS__)
 
+#define ASSERT_EQ(LHS, RHS) ASSERT((LHS) == (RHS), STR(LHS) " ({}) and " STR(RHS) " ({}) should be equal", (LHS), (RHS));
```

Comment:
These have a problem. They evaluate LHS and RHS multiple times. 

Use a macro with a do while loop in it to scope the variables because otherwise some things won't be able to be tested.